resources/assets/js/helpers/localize.ts

```diff
@@ -1,15 +1,27 @@
-import { localizedField } from "../models/app";
+import { localizedField, localizedFieldNonNull } from "../models/app";
 
 export type Locales = "en" | "fr";
-type LocalizedFields<T> = {
+type TranslatableKeysNonNull<T> = {
+  [K in keyof T]: T[K] extends localizedFieldNonNull ? K : never;
+}[keyof T];
+type TranslatableKeys<T> = {
   [K in keyof T]: T[K] extends localizedField ? K : never;
 }[keyof T];
 
 export function localizeField<T>(
   locale: Locales,
   model: T,
-  field: LocalizedFields<T>,
-) {
+  field: TranslatableKeys<T>,
+): string | null {
+  const a = model[field];
+  return model[field][locale];
+}
+export function localizeFieldNonNull<T>(
+  locale: Locales,
+  model: T,
+  field: TranslatableKeysNonNull<T>,
+): string {
+  const a = model[field];
```

Comment:
I think we'll want to do something with these `const a = model[field];`'s, either returning `a[locale]` or removing the `const` declaration.